src/RectEngine.cpp

```diff
@@ -417,7 +419,7 @@ RectEngine::warpPersceptive(void* src,  const std::vector<int>& poly2Imgs, const
     int outBatchSize = poly2Imgs.size();
     uchar* src_data =  static_cast<uchar*>(src);
     uchar* dst_data =  static_cast<uchar*>(buffer_mgr.mDeviceBuffer[mPerspectiveOutputBufferDevIdx].data());
-    float* dst_gray_data =  static_cast<float*>(buffer_mgr.mDeviceBuffer[mOutputBufferIndex].data());
+    float* dst_gray_data =  static_cast<float*>(buffer_mgr.mDeviceBuffer[mGrayOutputBufferDevIdx].data());
```

Comment:
If you only need RGB input, why still run the kernel to convert RGB to gray ?

Reply:
convert RGB to gray is used for previous OCR version, Yes, I can add a flag here to return RGB early as gray data is not needed for CLIP4STR

Reply:
Yes. And I'll suggest to initialize buffer in OCRNetEngine as it is binding to the input shape of the TRT engine. And you could figure out the number of channel from that shape in RectEngine and determine to run RGB process or Grayscale process.